src/main/java/Parser.java

```diff
@@ -0,0 +1,150 @@
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+/**
+ * Makes sense of the user command
+ */
+public class Parser {
+
+    /**
+     * Returns the response to the user's input
+     * @param input input supplied by the user
+     * @param ui    user interface in charge of displaying the responses to the user
+     * @throws DukeException if user command is invalid
+     */
+    public static String handleUserCommand(String input, Ui ui) throws DukeException {
+        if (input.equals("bye")) {
+            Duke.isFinished = true;
+            return ui.showGoodbyeMessage();
+        } else if (input.equals("list")) {
+            return ui.showTaskList();
+        } else if (input.startsWith("find")) {
+            String keyword = getKeyword(input);
+
+            ArrayList<String> matchedTasks = TaskList.findMatchingTasks(keyword);
+
+            return ui.showArrayList(matchedTasks);
+        } else if (input.startsWith("done ")) {
+            Task finishedTask = getFinishedTask(input);
+
+            TaskList.setDone(finishedTask);
+
+            return ui.showMessage("Nice! I've marked this task as done:" + finishedTask);
+        } else if (input.startsWith("todo ")) {
+            if (input.split(" ").length == 1) {
+                throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+            }
+            String todoTask = getTodoTask(input);
+
+            TaskList.createTodoTask(todoTask);
+
+            return ui.showMessage("added!\n" + "you have " + TaskList.updatedTaskList.size()
+                    + " tasks in your list");
+        } else if (input.startsWith("event ")) {
+            String eventTask = getEventTask(input);
+            String date = getEventDate(input);
+
+            TaskList.createEventTask(eventTask, date);
+
+            return ui.showMessage("added!\n" + "you have " + TaskList.updatedTaskList.size()
+                    + " tasks in your list");
+        } else if (input.startsWith("deadline ")) {
+            String deadlineTask = getDeadlineTask(input);
+            LocalDate deadline = getDeadlineDate(input);
+
+            TaskList.createDeadlineTask(deadlineTask, deadline);
+
+            return ui.showMessage("added!\n" + "you have " + TaskList.updatedTaskList.size()
+                    + " tasks in your list");
+        } else if (input.startsWith("delete ")) {
+            int taskNumber = getTaskNumberToBeDeleted(input);
+            Task taskDeleted = getTaskToBeDeleted(taskNumber);
+
+            TaskList.deleteTask(taskNumber);
+
+            return ui.showMessage("Noted. I've removed this task: " + taskDeleted + "\n"
+                    + "you have " + TaskList.updatedTaskList.size() + " tasks in your list");
+        } else {
+            throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    public static String getKeyword(String input) {
```

Comment:
Perhaps you could consider adding JavaDoc comments to describe what this public method does?